intercom/url.go

```diff
@@ -0,0 +1,19 @@
+package intercom
+
+import "github.com/amp-labs/connectors/common/urlbuilder"
+
+// Intercom pagination cursor sometimes ends with `=`.
+var intercomQueryEncodingExceptions = map[string]string{ //nolint:gochecknoglobals
+	"%3D": "=",
```

Comment:
Example: 
```
    "next": {
      "page": 2,
      "starting_after": "WzE3MTU2OTU2NzkwMDAsIjY2NDM3MDNmZmFlNzgzNGQxNzkyZmQzMCIsMl0="
    },
```
`starting_after` will be used as query parameter and it has `=` symbol. Sometimes it doesn't.

Reply:
looks like base64 data. Decoding it shows a value of `[1715695679000,"6643703ffae7834d1792fd30",2]`. Guessing it's ok to treat this as an opaque value though?

Reply:
Yes, it is.
I implemented cursor based pagination before. The implementation requires a cursor to be opaque string with the following properties:
* Ordered (this acts as a pointer in history. Before/after is robust against rapid modifications like news feed, avoiding stuttering, slipage when switching pages in offset style)
* Totally unqiue (this breaks ties if something happend at exact moment)
Not all tables/documents have a field that falls under both categories. First number is ordered, second is unique, third is unclear what it means - Intercom does composite approach.